Fix `woocommerceTranslation` used in title breadcrumb

tomalec opened the pull request:
### Changes proposed in this Pull Request:

- Rename `wc-admin-settings` to `settings` to match Woo/DEWP mapping. (86c2a76)
  Mention findings from https://github.com/woocommerce/woocommerce-admin/issues/7810- the unobvious source of unpublished `@woocommerce/settings` -> https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/assets/js/settings/shared/index.ts
- Fix `woocommerceTranslation` used in title breadcrumb (947aa89)
  Fixes #1050


### Screenshots:

<!--- Optional --->


### Detailed test instructions:

1. Checkout the branch
2. `npm install`
3. `npm run build`
4. Open any GLA page, like[`/wp-admin/admin.php?page=wc-admin&subpath=%2Freconnect-accounts&path=%2Fgoogle%2Fsettings`](https://gla1.test/wp-admin/admin.php?page=wc-admin&subpath=%2Freconnect-accounts&path=%2Fgoogle%2Fsettings)
5. Check the page title, `document.title` in the console


<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with `(Fix|Add|Tweak|Update) - `, for example:
> Fix - I took care of something that wasn't working.
> Add - I added something new that's pretty cool.
> Tweak - I made a small change.
> Update - I made big changes to something that wasn't broken.

Leave the "Changelog entry" header in place completely empty, without any summary if no changelog entry is needed.
If you remove the "Changelog entry" header, the title of Pull Request will be used as the changelog entry.
-->
### Changelog entry

> Fix - `woocommerceTranslation` used in title breadcrumb

tomalec committed:
Rename `wc-admin-settings` to `settings` to match Woo/DEWP mapping.

Mention findings from https://github.com/woocommerce/woocommerce-admin/issues/7810- the unobvious source of unpublished `@woocommerce/settings` -> https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/assets/js/settings/shared/index.ts

tomalec committed:
Fix `woocommerceTranslation` used in title breadcrumb

Fixes https://github.com/woocommerce/google-listings-and-ads/issues/1050

eason9487 commented on webpack.config.js:
```diff
@@ -15,14 +16,21 @@ const requestToExternal = ( request ) => {
+		// Please note, that this is NOT https://www.npmjs.com/package/@woocommerce/settings,
+		// or https://github.com/woocommerce/woocommerce-admin/tree/main/packages/wc-admin-settings
+		// but https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/assets/js/settings/shared/index.ts
+		// (at an unknown version).
```
💅 
According to [this composer package info](https://packagist.org/packages/woocommerce/woocommerce-blocks), which links the [woocommerce/woocommerce-gutenberg-products-block](https://github.com/woocommerce/woocommerce-gutenberg-products-block) repo, the version refers to here https://github.com/woocommerce/woocommerce/blob/5.8.0/composer.json#L25

tomalec replied:
What makes the version unknown is the use of DEWP itself.

eason9487 commented on webpack.config.js:
```diff
@@ -32,7 +40,7 @@ const requestToHandle = ( request ) => {
 		'@woocommerce/components': 'wc-components',
 		'@woocommerce/navigation': 'wc-navigation',
-		'@woocommerce/wc-admin-settings': 'wc-settings',
+		'@woocommerce/wc-settings': 'wc-settings',
```
🚧 
The key in `wcHandleMap` needs to be the same as the importing module name (`@woocommerce/settings`) used in js files. Otherwise, it will result in a blank page due to the `@woocommerce/settings` doesn't match any handle of WordPress script dependencies.

eason9487 commented on webpack.config.js:
```diff
@@ -77,9 +85,12 @@ const webpackConfig = {
 				plugin.constructor.name !== 'DependencyExtractionWebpackPlugin'
 		),
 		new DependencyExtractionWebpackPlugin( {
+			combineAssets: false,
```
❔ 
The default value of `combineAssets` is `false` already. There should be no need to specify, right?

eason9487 commented on webpack.config.js:
```diff
@@ -77,9 +85,12 @@ const webpackConfig = {
 		),
 		new DependencyExtractionWebpackPlugin( {
+			combineAssets: false,
+			outputFormat: 'json',
```
🚧 

This change breaks the importing asset file path:
https://github.com/woocommerce/google-listings-and-ads/blob/947aa89679fd750570bb16b5884d973cca0ca09e/src/Admin/Admin.php#L102

And the `$build_dependency_path` parameter doesn't support the asset file in JSON format:
https://github.com/woocommerce/google-listings-and-ads/blob/947aa89679fd750570bb16b5884d973cca0ca09e/src/Assets/AdminScriptWithBuiltDependenciesAsset.php#L56-L62
https://github.com/woocommerce/google-listings-and-ads/blob/947aa89679fd750570bb16b5884d973cca0ca09e/src/Value/ArrayWithRequiredKeys.php#L36-L39

tomalec replied:
Sorry and thanks for catching that. I committed by accident the changes I used to test https://github.com/WordPress/gutenberg/pull/35106

eason9487 commented on webpack.config.js:
```diff
@@ -77,9 +85,12 @@ const webpackConfig = {
 			injectPolyfill: true,
 			requestToExternal,
 			requestToHandle,
+			externalizedReportFile: 'externalized.json',
```
❔ 
I cannot find this option. What is it used for?

tomalec committed:
Fix name in the `wc-settings` "module" specifier.

Addresses https://github.com/woocommerce/google-listings-and-ads/pull/1055#discussion_r731493447

tomalec committed:
Remove unrelated DEWP experiments.

Addresses https://github.com/woocommerce/google-listings-and-ads/pull/1055#discussion_r731500951

tomalec commented:
@eason9487 Thanks for the review :) would you mind another round?

eason9487 reviewed (approved):
Tested with WC 5.5.1 and 5.8 and works well. LGTM.
